### ISSUE TITLE
undefined-loop-variable in presence of cell-var-from-loop

### ISSUE THREAD

**Issue by sk-** [PyCQA/pylint#2121]
Similar to bug #1921 and related, I found that an extra `undefined-loop-variable` error is shown  in presence of `cell-var-from-loop` when the same variable is used later in another loop.

### Steps to reproduce
The following code:
```py
def call(f, a):
    return f(a)
for x in [1]:
    call(lambda a: a+x, 3)
for x in [2]:
    pass
```

outputs

```
W:  4,21: Cell variable x defined in loop (cell-var-from-loop)
W:  4,21: Using possibly undefined loop variable 'x' (undefined-loop-variable)
```

However, when changing the name of the second `x` or commenting out those two lines, then the output is just 

```
W:  4,21: Cell variable x defined in loop (cell-var-from-loop)
``` 

### pylint --version output
```
No config file found, using default configuration
pylint 1.8.4,
astroid 1.6.3
Python 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 03:03:55)
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
```

**Comment by PCManticore** [comment]
Thanks for the bug report!

**Comment by SamyCookie** [comment]
Does https://github.com/PyCQA/pylint/pull/2276 fixed this issue ?

**Comment by Pierre-Sassoulas** [comment]
Closing as it's fixed on main, thank you for the help @SamyCookie  and @david-yz-liu
